components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -17,6 +20,13 @@ import (
 	"github.com/mitchellh/hashstructure"
 )
 
+// MessagingBackendHandler exposes a common handler interface for different messaging backend systems
+type MessagingBackendHandler interface {
+	Initialize(cfg env.Config) error
+	SyncSubscription(subscription *eventingv1alpha1.Subscription, cleaner eventtype.Cleaner, params ...interface{}) (bool, error)
```

Comment:
Any chance to have a different type than `interface{}` for the parameters? Or which is the concrete reason for it? Also could we please document the intention of each interface method as they are public and implementations have to fulfill them?

Reply:
I agree with @themue. Having variadic parameters of type interface{} makes it really hard to use correctly without reading the code. 

i see that we call `syncBEBSubscription` in the actual reconcile method. This one calls `SyncSubscription`. Would it make sense to use the new interface method `SyncSubscription` in the reconciler directly. That way all backend specific functionality would be hidden within the interface implementation

Reply:
We have different parameters for different backends. For example, this is the call for BEB susbcription synchronization:
```
if statusChanged, err = r.bebClient.SyncSubscription(subscription, r.eventTypeCleaner, apiRule); err != nil {
```
where 
```
apiRule *apigatewayv1alpha1.APIRule
```
It's a compromise for now. The client of this interface and the implementation of it must to be in sync regarding the variadic parameters

Reply:
@k15r ```syncBEBSubscription``` is a private method of the current BEB controller, implementing a part of the BEB controller specific logic. This will change later when we will unify the controller logic. Till now,  only the working with the backend system is abstracted in a common interface.

Reply:
Comments added to each method of the interface

Reply:
Having a small PR which could be merged right now could be better for the current situation as to have a huge PR for the complete "unification" of controllers. This is why this PR tries to solve only the functionality related to the "communication" with the backend system.